### ISSUE TITLE
[AutoML v0.16.0] InferColumn doesn't work on tricky csv file

### ISSUE THREAD

**Issue by LittleLittleCloud** [dotnet/machinelearning#4460]
For some csv file that contains double quotes in it's field, the `inferColumn` API can't work properly. It's probably because when guessing delimiter, AutoML takes the candidates inside double quote into consideration, which should be neglect. (Or when splitting lines, it uses \n inside double quote)

steps to reproduce:
download this [dataset](http://data.insideairbnb.com/the-netherlands/north-holland/amsterdam/2019-09-14/visualisations/listings.csv)

```
MLContext mlContext = new MLContext();
var inputColumnInformation = new ColumnInformation();
inputColumnInformation.LabelColumnName = @"review_scores_rating";
var train = mlContext.Auto().InferColumns(TrainDataPath, inputColumnInformation);
```

#### Updated
The dataset actually works for latest AutoML/ModelBuilder, To reproduce the error, please uses this dataset:

[jigsaw.txt](https://github.com/dotnet/machinelearning/files/4631129/jigsaw.txt)

**Comment by justinormont** [comment]
This is a issue of the TextLoader in ML.NET. It does not currently support escaped quotes in a quoted field.

The TextLoader has a rather limited support for TSV/CSV files. 

The issue is noted in the old repo: https://github.com/dotnet/machinelearning-automl/issues/193 ("Infercolumn fails to parse new lines inside quoted text")

> 
> ---
> 
> > @vinodshanbhag : 
> > Wikidetox fails in benchmarking because of this.
> > Tools like Excel are able to handle this.
> 
> ---
> 
> 
> > @justinormont :
> > @CESARDELATORRE: what do you think about writing an example of converting a dataset from CSV/TSV to IDV?
> > 
> > The `TextLoader` can not handle many CSV/TSV files. Using a more general reader and outputting to IDV would the allow the AutoML code to read the IDV format. 
> > 
> > Basic example:
> > * General (C# CSV/TSV) reader which returns an `IEnumerable` (maybe [CsvHelper](https://joshclose.github.io/CsvHelper/))
> > * ML.NET creates an `IDataView` from the `IEnumerable` https://github.com/dotnet/machinelearning/blob/d6a4b1dd79e13c9a749ad9fdeb1c6ea2f4f547a6/docs/code/SchemaComprehension.md#streaming-data-views
> > * ML.NET writes to an IDV/TDV file -- `mlContext.Data.SaveAsBinary("pathToFile")`
> > 
> > To be clear, this would be an example (docs/example code) of how a user could convert their data before it comes to AutoML. This would allow us to process files like this issue is referencing. 
> 
> ---
> 
> > @CESARDELATORRE :
> > @justinormont - It's a good idea. However, This example should be a workaround for cases like that.
> > It might also be a good example because those issues with `"numeric value"` happen in ML.NET 0.11 per-se.
> > 
> > For instance, I was using another dataset yesterday (just migrating to ML.NET v0.11) where the column `Label` had values like:
> > 
> > - `"1"`
> > - `"0"`
> > 
> > ML.NET transformers were not able to convert that to Boolean (it was putting all as 0) neither to Float (all values as NaN)… See issue in ML.NET I created: 
> > 
> > https://github.com/dotnet/machinelearning/issues/2824
> > 
> > Interestingly, those conversions to Boolean were working properly until ML.NET v0.10...
> > 
> > So, yes, this can be a good example. However, for AutoML, this example should be a workaround. For most cases, a .CSV/TSV files should be the by default approach since that is the most common type of dataset.
> 
> ---
> 
> > @justinormont :
> > @CESARDELATORRE - This is a side-effect of turning off quoting by default in ML.NET: 
> > https://github.com/dotnet/machinelearning/pull/2630
> > 
> > Non-issue:
> > I _think_ AutoML will be unaffected by ML.NET changing its quoting defaults, as we sweep over both choices (and our heuristics default to using quoting when all else is equal). We should verify.
> > 
> > Issue:
> > AutoML will be affected by the `TextLoader` not supporting common TSV/CSV files. The purposed work around above is telling a user how to convert their TSV/CSV to IDV (bypassing `TextLoader`).
> 
>

**Comment by LittleLittleCloud** [comment]
Good to know about it
Thanks!

**Comment by harishsk** [comment]
@LittleLittleCloud I am assuming your question has been answered and am closing the issue. Please feel free to reopen the issue if you have more questions.

**Comment by justinormont** [comment]
We may want to open an issue to improve the TextLoader to support more common TSV/CSV formats.

**Comment by LittleLittleCloud** [comment]
Is there a way to bypass the restriction in AutoML? like provide an internal columnInference API using TextReader or something.

I can help investigate that, ModelBuilder has already done something handling escaped quotes inside quotes, maybe we can use that @justinormont @JakeRadMSFT

**Comment by JakeRadMSFT** [comment]
@LittleLittleCloud lets give that a shot. Can you open an issue on Model Builder side?

**Comment by antoniovs1029** [comment]
So as discussed offline with @LittleLittleCloud :
1. The issue he mentioned with CSV files that have new lines inside quoted fields is fixed in #5125 
2. We're not clear about what was the issue he mentioned regarding having double quotes (""). The file he shared as repro doesn't contain new lines inside quoted fields (so it is unaffected by #5125 ) but it contains double quotes. ML.NET's Textloader has always been capable of loading CSV files with double quotes, so there's no issue in Textloader regarding this.

**Comment by antoniovs1029** [comment]
PR #5125 has fixed the problem in the TextLoader to load new line characters as part of quoted fields, as part of a new option called readMultilines.

As mentioned here: https://github.com/dotnet/machinelearning/pull/5125#discussion_r426017274 we'll leave it up to the ModelBuilder to decide how to surface this new option through AutoML, and if they actually want to expose this through InferColumn() to properly fix the issue that was created here by @LittleLittleCloud .

**Comment by antoniovs1029** [comment]
So this has been fixed by #5125 and #5148 😄
